Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-apps.ts

```diff
@@ -509,8 +509,18 @@ export class AzureSpringApps {
      */
     private async applyDeploymentModifications(appName: string, deploymentName: string, deploymentUpdateRequestBody, createDeployment: boolean) {
         console.log(`${createDeployment ? 'Creating' : 'Updating'} ${appName}, deployment ${deploymentName}...`);
-
-        let method = createDeployment ? 'PUT' : 'PATCH';
+        let isCustomContainer = false;
+        let deploymentResource = {};
+        if (!createDeployment) {
+            deploymentResource = await this.getDeploymentInfo(appName, deploymentName);
+            isCustomContainer = deploymentResource["properties"]["source"]["type"] == SourceType.CUSTOM_CONTAINER || deploymentUpdateRequestBody["properties"]["source"]["type"] == SourceType.CUSTOM_CONTAINER;
+        }
+        let method = (createDeployment || isCustomContainer) ? 'PUT' : 'PATCH';
```

Comment:
per talk offline, please also check if we need to update for other source types.

Reply:
I have tried Jar/Source in Standard tier and I can use PATCH to deploy Source to a Jar app.
For custom container cases, we have a validator logic in RP deployment controller to reject the PATCH request about byoc cases and throw `properties.source.type can only be changed in PUT request` message